src/lexer/tokenizer.c

```diff
@@ -0,0 +1,85 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   tokenizer.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: lrocigno <lrocigno@student.42sp.org.br>    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2022/02/01 13:49:35 by lrocigno          #+#    #+#             */
+/*   Updated: 2022/02/03 07:16:44 by vgoncalv         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <lexer/lexer.h>
+
+/**
+ * Count the amount of tokens to create a array of them.
+ */
+
+static size_t	cnt_tokens(char *input)
+{
+	size_t	cnt;
+	size_t	i;
+	int		quotes;
+	int		is_word;
+
+	cnt = 0;
+	i = 0;
+	quotes = 0;
+	is_word = 0;
+	while (input[i] != '\0' && input[i] != '#')
+	{
+		if (ft_isalnum(input[i]) && !is_word)
+		{
+			++cnt;
+			is_word = 1;
+		}
+		else if (lex_isspace(input[i]) && quotes % 2 == 0)
+			is_word = 0;
+		else if (input[i] == '\"' || input[i] == '\'')
+			++quotes;
```

Comment:
Isso introduz um problema: no Minishell, quando tiver uma aspas dupla ou simples, e ela não for terminada, é para considerá-la como um caractere apenas. Vou mudar o fluxo para poder embarcar todos os casos:

1. Transformar o t_token numa lista ligada(evitamos ter de contar os tokens antes de tê-los, eliminando o problema supracitado)
2. A checagem no loop principal em tokenizer passa a ser se foi criado um token ou não (dessa forma, quando o input for vazio, não há tokens)